src/ir/value_test.cc

```diff
@@ -118,14 +119,14 @@ TEST_P(ValueToStringTest, ToStringReturnsExpectedFormat) {
 INSTANTIATE_TEST_SUITE_P(
     ValueToStringTest, ValueToStringTest,
     testing::Values(
-        std::make_pair(Value(test_data.first_block_first_arg), "%0.one"),
-        std::make_pair(Value(test_data.first_block_second_arg), "%0.two"),
-        std::make_pair(Value(test_data.second_block_first_arg), "%1.one"),
-        std::make_pair(Value(test_data.second_block_second_arg), "%1.two"),
-        std::make_pair(Value(test_data.plus_operation_result1), "%0.res1"),
-        std::make_pair(Value(test_data.plus_operation_result2), "%0.res2"),
-        std::make_pair(Value(test_data.minus_operation_result1), "%1.res1"),
-        std::make_pair(Value(test_data.minus_operation_result2), "%1.res2"),
+        std::make_pair(Value(test_data.first_block_first_arg), "%0"),
```

Comment:
You should populate SSA names so that we don't rely on the order in which names were assigne.d 

Also, this is somewhat related to #596 . Will need to figure out how to refer to block arguments and results in the parser.

Reply:
The reason ssa names reply on the order in which names are assigned is because of the way we populate ssa names based on the size of the hash map. I think we do this to make sure that ids are unique. To avoid relying on order, specifically for Values, we can do some combination of hash map size and value name. (this also preserves uniqueness of ids)

Reply:
Moving Value changes and discussions to another PR

Reply:
Because I had time before the another round of review, added changes to value test.